### DISCUSSION TITLE
Diarization not working for pre-recorded files

### DISCUSSION THREAD

**Discussion by vincentpaca** [deepgram/community#611] [General help]
Hi,

I've been using both the playground and the API to diarize a fairly short audio snippet of 2 synthesized voices talking. 

Enabling `diarize` does not work both in the playground, as well as just calling the API directly in my app.

I have tried to run it will all the different available models and none of them work with diarziation.

I also tried to diarize using other providers and they can identify speakers just fine with the same audio snippet.

What is even more curious is if I use streaming, the diarization works with the `base` model. I have the mic streaming to the websocket endpoint and just play the same audio separately, Deepgram gives me the diarzed transcript just fine.

Anyone else having the same issues? I'm stumped.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by martapernas** [reply]
Did you get an answer to this ? I had the same issue. I am currently testing a recorded audio file via the api playground, and I noticed that it could get the different speakers.
Could you please provide any updates on this ?

Thanks,

**Comment by DamienDeepgram** [comment]
> Hi,
> 
> I've been using both the playground and the API to diarize a fairly short audio snippet of 2 synthesized voices talking.
> 
> Enabling `diarize` does not work both in the playground, as well as just calling the API directly in my app.
> 
> I have tried to run it will all the different available models and none of them work with diarziation.
> 
> I also tried to diarize using other providers and they can identify speakers just fine with the same audio snippet.
> 
> What is even more curious is if I use streaming, the diarization works with the `base` model. I have the mic streaming to the websocket endpoint and just play the same audio separately, Deepgram gives me the diarzed transcript just fine.
> 
> Anyone else having the same issues? I'm stumped.

Could you provide the sample audio you used and the API params or a Request ID so we can look into this further please?

**Reply by vincentpaca** [reply]
Hi Damien, here's the sample audio: https://drive.google.com/file/d/1NzB1WRJzXq5vNUwyCtu33TiaHJkr6kwZ/view?usp=drive_link

Using the following params:

`smart_format=true&diarize=true`

The workaround right now is to use the websocket endpoint, but it's not ideal.

**Reply by martapernas** [reply]
Did you get an answer to this ? I had the same issue. I am currently testing a recorded audio file via the api playground, and I noticed that it could get the different speakers. 
Could you please provide any updates on this ?

Thanks, 
fcisco

**Reply by vincentpaca** [reply]
Hi @martapernas , no official answer from the team but currently using a workaround.

If you use the WebSocket endpoint instead of the HTTPs endpoint for your pre-recorded audio, the diarization should work.

Although now I'm noticing a new issue... the first few (1-3) spoken words of the 2nd speaker always gets tagged as the first speaker for some reason. This is repeatable for my audio samples (will make a new ticket for this as it can be a separate bug).

**Reply by martapernas** [reply]
Thanks a lot for your feedback. We tested using youtube videos and it worked find, but when testing with inhouse audio recordings, no diarization or speaker recognition. It could not recognize the speakers.

**Reply by vincentpaca** [reply]
Any update @DamienDeepgram ? I think this is a bug.

**Reply by DamienDeepgram** [reply]
Thanks for the sample file I was able to reproduce and shared with the team.

**Reply by SoyKevin111** [reply]
fixed?
